python/sglang/srt/layers/communicator.py

```diff
@@ -264,6 +269,43 @@ def should_use_reduce_scatter(self, forward_batch: ForwardBatch):
             and forward_batch.dp_padding_mode.is_max_len()
         )
 
+    def should_fuse_mlp_allreduce_with_next_layer(
+        self, forward_batch: ForwardBatch
+    ) -> bool:
+        speculative_algo = global_server_args_dict.get("speculative_algorithm", None)
+        if (
+            is_dp_attention_enabled()
+            and speculative_algo is not None
+            and hasattr(speculative_algo, "is_eagle")
```

Comment:
qq: it seems speculative_algo is of type SpeculativeAlgorithm, thus should always have this attr, thus maybe we can remove this hasattr

Reply:
Done!